src/PythonMigrationViewExtension/MigrationAssistant/migrate_2to3.py

```diff
@@ -0,0 +1,46 @@
+from lib2to3.pgen2.parse import ParseError
+from lib2to3.refactor import RefactoringTool, get_fixers_from_package
+import zipfile
+import os.path
+from pathlib import Path
+import sys
+
+def transform(source):
+    python_zip_path = str(Path(os.path.dirname(os.__file__)).parents[0])
+    python_zip_file = get_zip_file(python_zip_path)
+    python_zip_folder = zipfile.ZipFile(os.path.join(python_zip_path, python_zip_file))
+    fixers = get_all_fixers_from_zipfolder(python_zip_folder)
+
+    refactoring_tool = RefactoringTool(fixers)
+
+    return refactor_script(source, refactoring_tool)
+
+def refactor_script(source_script, refactoring_tool):
+    # lib2to3 likes a newline at the end
+    source_script += '\n'
+    try:
+        tree = refactoring_tool.refactor_string(source_script, 'script')
+    except ParseError as e:
+        if e.msg != 'bad input' or e.value != '=':
+            raise
+        tree = refactoring_tool.refactor_string(source_script, self.pathname)
+    return str(tree)[:-1] # remove added newline 
+
+def get_zip_file(path):
+    zip_file = ""
+    for f in os.listdir(path):
+        if f.startswith("python") and f.endswith(".zip"):
+            zip_file = f
+            break
+    return zip_file
+
+def get_all_fixers_from_zipfolder(folder):
```

Comment:
if this is our `py` code and not the official one or something like that, we should add some comments to this explaining 
- what all the zipping is about
- what these `fixers` are

Are the fixers in a `zip` file ?

Reply:
yea this is ours. `fixers` are what 2to3 uses to check your code, essentially they are small python classes, you can build your own or you can uses 2to3's OOTB fixers. Currently we are using the OOTB fixers as they should be enough. Normally you would get the OOTB fixers by doint `get_fixers_from_package('lib2to3.fixes')`, but becasue we are using an embedded Python environment which means that the environment is zipped, we cant do it like that (this is a known limitation with 2to3). Insted i am navigating to the embedded environment and getting all the fixers from that zip file.